tests/unit/test_query_language.py

```diff
@@ -179,6 +185,32 @@ def test_year(self):
         )
 
 
+@pytest.mark.parametrize(
+    "lhs,op,rhs,expected_type",
+    [
+        (patients.f, "-", 10, FloatPatientSeries),
+        (patients.f, "+", 10, FloatPatientSeries),
+        (patients.f, "<", 10, BoolPatientSeries),
+        (events.f, "-", 10, FloatEventSeries),
+        (events.f, "<", 10, BoolEventSeries),
+        (events.f, ">", 10, BoolEventSeries),
+        (events.f, "<", 10.0, BoolEventSeries),
```

Comment:
I'm not sure this covers all cases and am also not sure if the last one is needed.

Reply:
I think the coverage here is fine.

Because of the way `_cast()` works we only have to test a single float operation with a single integer value to get 100% coverage. But I think it's useful to have a few more examples here, as you've done, because these tests are fast to run and the parameterisation means the examples themselves are very short and readable.

Reply:
great, thanks for explaining